### PR TITLE
test(x/ecocredit): fix intermittent race condition

### PR DESCRIPTION
**Pull request by ryanchristo**
## Description

This pull request fixes intermittent race condition caused by random string generator in tests:

```
  github.com/cosmos/cosmos-sdk/types/simulation.RandStringOfLength()
      /home/runner/go/pkg/mod/github.com/regen-network/cosmos-sdk@v0.45.0-regen-1/types/simulation/rand_util.go:29 +0x114
  github.com/regen-network/regen-ledger/x/ecocredit/core.TestMsgCreateClass()
      /home/runner/work/regen-ledger/regen-ledger/x/ecocredit/core/msg_create_class_test.go:92 +0xc0d
```

```
  github.com/cosmos/cosmos-sdk/types/simulation.RandStringOfLength()
      /home/runner/go/pkg/mod/github.com/regen-network/cosmos-sdk@v0.45.0-regen-1/types/simulation/rand_util.go:27 +0x7c
  github.com/regen-network/regen-ledger/x/ecocredit/core.TestMsgCreateBatch()
      /home/runner/work/regen-ledger/regen-ledger/x/ecocredit/core/msg_create_batch_test.go:235 +0x1f6d
```

---

### Author Checklist

*All items are required. Please add a note to the item if the item is not applicable and
please add links to any relevant follow up issues.*

I have...

- [ ] included the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] added `!` to the type prefix if API or client breaking change
- [ ] targeted the correct branch (see [PR Targeting](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] provided a link to the relevant issue or specification
- [ ] followed the guidelines for [building modules](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules)
- [ ] included the necessary unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] added a changelog entry to `CHANGELOG.md`
- [ ] included comments for [documenting Go code](https://blog.golang.org/godoc)
- [ ] updated the relevant documentation or specification
- [ ] reviewed "Files changed" and left comments if necessary
- [ ] confirmed all CI checks have passed

### Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] confirmed the correct [type prefix](https://github.com/commitizen/conventional-commit-types/blob/v3.0.0/index.json) in the PR title
- [ ] confirmed `!` in the type prefix if API or client breaking change
- [ ] confirmed all author checklist items have been addressed 
- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] reviewed documentation is accurate
- [ ] reviewed tests and test coverage
- [ ] manually tested (if applicable)

### CODE REVIEW

**Comment by ryanchristo** [comment]
For example, just hit this error again: https://github.com/regen-network/regen-ledger/runs/6014956628?check_suite_focus=true

File: x/ecocredit/core/msg_create_batch_test.go [hunk 1]
```diff
@@ -232,7 +231,7 @@ func TestMsgCreateBatch(t *testing.T) {
 				ProjectId: "C01",
 				StartDate: &startDate,
 				EndDate:   &endDate,
-				Metadata:  simtypes.RandStringOfLength(r, 288),
+				Metadata:  string(make([]byte, 288)),
```
Comment (robert-zaremba) [inline x/ecocredit/core/msg_create_batch_test.go:234]: we could create a a random string at the package level, and reuse it in tests.

Comment (ryanchristo) [reply x/ecocredit/core/msg_create_batch_test.go:234]: Not necessary here when all we are checking is if the metadata is too long.

**Review by technicallyty — APPROVED** [review]
LGTM

**Pull request merged**